assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -195,6 +199,70 @@ const baseActions = {
 			payload: { customDimensions: [] },
 		};
 	},
+
+	/**
+	 * Sets a schedule timeout ID for syncing available custom dimensions in state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {number} syncTimeoutID The timeout ID.
+	 * @return {Object} A redux-style action.
+	 */
+	setSyncTimeoutID( syncTimeoutID ) {
+		return {
+			payload: { syncTimeoutID },
+			type: SET_SYNC_TIMEOUT_ID,
+		};
+	},
+
+	/**
+	 * Schedules a sync of available custom dimensions in state.
+	 *
+	 * @since n.e.x.t
+	 */
+	*scheduleSyncAvailableCustomDimensions() {
+		yield {
+			payload: {},
+			type: SCHEDULE_SYNC_AVAILABLE_CUSTOM_DIMENSIONS,
+		};
+	},
+};
+
+export const baseControls = {
+	[ SCHEDULE_SYNC_AVAILABLE_CUSTOM_DIMENSIONS ]: createRegistryControl(
+		( { select, dispatch } ) =>
+			() => {
```

Comment:
This function could be a bit cleaner by destructuring action and selector functions out at the top.

Reply:
Updated, thanks!